core/cache/watcher.go

```diff
@@ -156,6 +156,7 @@ func (w *ConfigWatcher) configChanged(topic string, value interface{}) {
 		return
 	}
 	w.notify()
+	w.hash = hash
```

Comment:
This is the actual fix

Reply:
Did you play around with whether this should happen before or after the notify? (I think after is correct, but I wanted to make sure we thought about it.)

Reply:
Yeah I did, but the conclusion I came to was the same.